geomesa-filter/src/main/scala/org/locationtech/geomesa/filter/visitor/BoundsFilterVisitor.scala

```diff
@@ -0,0 +1,83 @@
+/***********************************************************************
+* Copyright (c) 2013-2016 Commonwealth Computer Research, Inc.
+* All rights reserved. This program and the accompanying materials
+* are made available under the terms of the Apache License, Version 2.0
+* which accompanies this distribution and is available at
+* http://www.opensource.org/licenses/apache2.0.php.
+*************************************************************************/
+
+package org.locationtech.geomesa.filter.visitor
+
+import com.vividsolutions.jts.geom.Envelope
+import org.geotools.filter.visitor.ExtractBoundsFilterVisitor
+import org.geotools.geometry.jts.ReferencedEnvelope
+import org.locationtech.geomesa.utils.geotools.wholeWorldEnvelope
+import org.opengis.filter._
+import org.opengis.filter.spatial._
+import org.opengis.filter.temporal._
+
+/**
+  * Helper for extracting bounds from a filter. For some reason, ExtractBoundsFilterVisitor does
+  * not accept a generic Filter...
```

Comment:
Ok, we might be using this filter visitor backwards.  See https://github.com/geotools/geotools/blob/master/modules/plugin/shapefile/src/main/java/org/geotools/data/shapefile/ShapefileFeatureSource.java#L323